libtenzir/builtins/formats/kv.cpp

```diff
@@ -146,6 +195,8 @@ class kv_parser final : public plugin_parser {
         // handle this.
         auto [head, tail] = field_split_.split(rest);
         auto [key, value] = value_split_.split(head);
+        key = trim_quotes(key);
+        value = trim_quotes(value);
```

Comment:
I didnt consider the effects on stripping when allowing an even quote count.

I think it will be best to just accept exactly 0 or two quotes during splitting.